src/propsToBody.js

```diff
@@ -106,4 +106,11 @@ const propsToBody = (uuid, props, type) => {
   return body
 }
 
+function castMaterial(material) {
+  if (material instanceof Material) {
```

Comment:
As I mentioned in the previous attempt at this, it will never be an instance of a material.

`propsToBody` takes the serialized parameters of cannon physics bodies and turns them into physics bodies.

Reply:
This may have previously been correct.

`propsToBody` is used in 2 places:

1️⃣  `Debug`: Outside of the web worker: https://github.com/pmndrs/use-cannon/blob/b62aef8cd30c2165d5a121d5f383392e221b903c/src/Debug.tsx#L67
Will not satisfy this conditional and should use https://github.com/pmndrs/use-cannon/blob/b62aef8cd30c2165d5a121d5f383392e221b903c/src/propsToBody.js#L113

If I understand this code correctly, this should be behave as it did before, as you described.

2️⃣  In the `worker` after `getMaterialForOptions` (🆕 ): https://github.com/pmndrs/use-cannon/blob/b62aef8cd30c2165d5a121d5f383392e221b903c/src/worker.js#L152-L153
- `getMaterialForOptions` converts from `MaterialOptions` to a `Material`: https://github.com/pmndrs/use-cannon/blob/b62aef8cd30c2165d5a121d5f383392e221b903c/src/worker.js#L53-L64
It memoizes returning the same `Material` instance when using the same material name: https://github.com/pmndrs/use-cannon/blob/b62aef8cd30c2165d5a121d5f383392e221b903c/src/worker.js#L55-L60

Therefore, with the 🆕  `getMaterialForOptions` called first the `material` property _becomes_ a `Material` instance before being passed into `propsToBody`.

> `propsToBody` takes the serialized parameters of cannon physics bodies and turns them into physics bodies.

To make this description _still_ true, I can move the memoization of the `Material` instances into the `propsToBody`.

For example (likely not the final signature):

```diff
- const propsToBody = (uuid, props, type) => {
+ const propsToBody = (uuid, props, type, getMaterialForOptions) => {
  ...
  const body = new Body({
    ...
-    material: material ? new Material(material) : undefined,
+    material: material ? getMaterialForOptions(material) : undefined,
  })
```

Reply:
@bjornstar does the above make sense? Anything I missed?

Please let me know if you have any more code review feedback so I can polish this PR for you. Thanks!

Reply:
I think it would be alright if we pass the memoizing function into propsToBody.

Reply:
Updated.